unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/AvatarModifierArea/Handler/AvatarModifierAreaComponentHandler.cs

```diff
@@ -11,200 +9,54 @@ namespace DCL.ECSComponents
-        {
-            if (entity?.gameObject == null)
-                return null;
-            
-            UnityEngine.Vector3 center = entity.gameObject.transform.position;
-            Quaternion rotation = entity.gameObject.transform.rotation;
-            return ECSAvatarUtils.DetectAvatars(boxArea, center, rotation, excludedColliders);
-        }
-
-        private void RemoveAllModifiers()
-        {
-            RemoveAllModifiers(DetectAllAvatarsInArea());
-            avatarsInArea.Clear();
+                removed = true
+            });
         }
 
-        private void RemoveAllModifiers(HashSet<GameObject> avatars)
+        public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBAvatarModifierArea model)
         {
-            if (avatars == null)
+            if (model.Modifiers.Count == 0)
                 return;
 
-            foreach (GameObject avatar in avatars)
-            {
-                OnAvatarExit?.Invoke(avatar);
-            }
-        }
+            var internalComponentModel = internalAvatarModifierArea.GetFor(scene, entity)?.model ?? new InternalAvatarModifierArea();
 
-        private void ApplyCurrentModel(PBAvatarModifierArea model)
-        {
-            dataStore.ownPlayer.OnChange -= OwnPlayerOnOnChange;
-            dataStore.otherPlayers.OnAdded -= OtherPlayersOnOnAdded;
+            internalComponentModel.area = ProtoConvertUtils.PBVectorToUnityVector(model.Area);
 
-            if (model.Modifiers.Count > 0)
+            foreach (AvatarModifierType modifierKey in model.Modifiers)
             {
-                // We set the unity engine Vector3 here, so we don't allocate a Vector3 each time we use it
-                boxArea = ProtoConvertUtils.PBVectorToUnityVector(model.Area);
-                
-                // Add all listeners
-                foreach (AvatarModifierType modifierKey in model.Modifiers)
-                {
-                    var modifier = factory.GetOrCreateAvatarModifier(modifierKey);
-                    
-                    OnAvatarEnter += modifier.ApplyModifier;
-                    OnAvatarExit += modifier.RemoveModifier;
-                }
-
-                // Set excluded colliders
-                excludedColliders = GetExcludedColliders(model);
-
-                if (model.ExcludeIds.Count > 0)
-                {
-                    dataStore.ownPlayer.OnChange += OwnPlayerOnOnChange;
-                    dataStore.otherPlayers.OnAdded += OtherPlayersOnOnAdded;
-                }
+                var modifier = factory.GetOrCreateAvatarModifier(modifierKey);
 
-                this.model = model;
-                
-                // Force update due to after model update modifiers are removed and re-added
-                // leaving a frame with the avatar without the proper modifications
-                Update();
+                internalComponentModel.OnAvatarEnter += modifier.ApplyModifier;
```

Comment:
Could we get duplicated subscriptions here?

Reply:
uhmm yes you are right, just in case I added those events cleanup before that iteration, thanks!